lib/mix/tasks/compile/live_view.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Mix.Tasks.Compile.LiveView do
+  use Mix.Task
+  @recursive true
+
+  alias Mix.Task.Compiler.Diagnostic
+
+  @doc false
+  def run(args) do
+    {compile_opts, _argv, _errors} = OptionParser.parse(args, strict: [return_errors: :boolean])
+
+    diagnostics = validate_components_calls(project_modules())
+
+    if !compile_opts[:return_errors] do
+      print_diagnostics(diagnostics)
+    end
+
+    {:ok, diagnostics}
+  end
+
+  @doc false
+  def validate_components_calls(modules) do
+    for module <- modules,
+        Code.ensure_loaded?(module),
+        function_exported?(module, :__components_calls__, 0),
+        call <- module.__components_calls__() do
+      [validate_required_attrs(call), validate_undefined_attrs(call)]
+    end
+    |> List.flatten()
+  end
+
+  defp validate_required_attrs(call) do
+    %{component: {module, fun}, attrs: call_attrs, file: file, line: line} = call
+    attrs_defs = module.__components__()[fun] || []
+    passed_attrs = Enum.map(call_attrs, &elem(&1, 0))
+
+    for %{name: name, opts: opts} <- attrs_defs, opts[:required], "#{name}" not in passed_attrs do
+      message = "missing required attribute `#{name}` for component `#{inspect(module)}.#{fun}/1`"
+      error(message, file, line)
+    end
+  end
+
+  defp validate_undefined_attrs(call) do
+    %{component: {module, fun}, attrs: call_attrs, file: file} = call
+    attrs_defs = module.__components__()[fun] || []
+    defined_attrs = Enum.map(attrs_defs, fn %{name: name} -> "#{name}" end)
+
+    for {name, _value, %{line: line}} <- call_attrs, name not in defined_attrs do
+      message = "undefined attribute `#{name}` for component `#{inspect(module)}.#{fun}/1`"
+      error(message, file, line)
+    end
+  end
+
+  defp project_modules do
+    files =
+      Mix.Project.config()[:app]
+      |> Application.app_dir()
+      |> Path.join("ebin")
+      |> File.ls!
+      |> Enum.sort()
+
+    for file <- files, {basename, ".beam"} <- [String.split_at(file, -5)] do
+      String.to_atom(basename)
+    end
```

Comment:
Note for the future: we want to cache this. The idea is that here we will load all modules and get only the ones that define a `__component_calls__` function. Then, we will save them to a file. We will only recompute this file if any of the files in `Mix.Tasks.Compile.Elixir.manifests()` is more recent than the file we created.

Reply:
Actually, this cache would be silly, because the only time we need to run these checks is if the Compile.Elixir.manifests() fail. Ignore this for now.